vm_args.c

```diff
@@ -856,7 +860,15 @@ setup_parameters_complex(rb_execution_context_t * const ec, const rb_iseq_t * co
             args_setup_kw_parameters_from_kwsplat(ec, iseq, keyword_hash, klocals, remove_hash_value);
         }
         else {
-            VM_ASSERT(args_argc(args) == 0);
+            VM_ASSERT(args_argc(args) == 0 ||
```

Comment:
It is better to use

```C
  if (args_argc(args) != 0) {
    VM_ASSERT(cond1);
    VM_ASSERT(cond2);
    ...
  }
```

because it is hard to read which assumption violates.

Reply:
I'll make that change.